pkgs/tools/text/excelCompare/default.nix

```diff
@@ -0,0 +1,28 @@
+{stdenv, lib, fetchurl, makeWrapper, unzip, jdk}:
+
+stdenv.mkDerivation rec {
+  name = "ExcelCompare-${version}";
+  version = "0.6.1";
+  src = fetchurl {
+    url = "https://github.com/na-ka-na/ExcelCompare/releases/download/${version}/ExcelCompare-${version}.zip";
+    sha256 = "1304qraazqm8mgjgnijzl070l0haa8jds7spnsy3xwh3vda0ka4x";
+  };
+  buildInputs = [ jdk unzip makeWrapper ];
+  sourceRoot = ".";
+
+  installPhase = ''
+    mkdir -p $out
+    cp -r bin $out/bin
+    chmod 555 $out/bin/excel_cmp
+    mv $out/bin/excel_cmp $out/bin/excel_cmp.sh
+    makeWrapper $out/bin/excel_cmp.sh $out/bin/excel_cmp \
+      --prefix PATH : ${lib.makeBinPath [ jdk ]}
```

Comment:
you don't need to rename excel_cmp before wrapping: `wrapProgram` is an equivalent of `makeWrapper` which wraps in place.
But reading the script it would be better to use the approach of https://nixos.org/nixpkgs/manual/#sec-language-java
that is:
* put all jars in $out/share/excel-cmp
* wrap java exactly as they do in the example, setting the class path to $out/share/excel-cmp/*

Reply:
Thanks for the review. In the example for makeWrapper, what am I supposed to put in place of `org.foo.Main`? I tried `com.ka.spreadsheet.diff.SpreadSheetDiffer` as is in the original `excel_cmp` startup script, but it results in `Error: Could not find or load main class .nix.store.3ykvmyv0ihvdpvxvdcivld3jf89blj83-ExcelCompare-0.6.1.share.excel_cmp.commons-digester-1.8.jar`.

Reply:
this works for me:
```diff
diff --git a/pkgs/tools/text/excelCompare/default.nix b/pkgs/tools/text/excelCompare/default.nix
index a0ec897eb5e..b433aa5e518 100644
--- a/pkgs/tools/text/excelCompare/default.nix
+++ b/pkgs/tools/text/excelCompare/default.nix
@@ -1,4 +1,4 @@
-{stdenv, lib, fetchurl, makeWrapper, unzip, jdk}:
+{stdenv, lib, fetchurl, makeWrapper, unzip, jre}:
 
 stdenv.mkDerivation rec {
   name = "ExcelCompare-${version}";
@@ -7,16 +7,14 @@ stdenv.mkDerivation rec {
     url = "https://github.com/na-ka-na/ExcelCompare/releases/download/${version}/ExcelCompare-${version}.zip";
     sha256 = "1304qraazqm8mgjgnijzl070l0haa8jds7spnsy3xwh3vda0ka4x";
   };
-  buildInputs = [ jdk unzip makeWrapper ];
+  nativeBuildInputs = [ unzip makeWrapper ];
   sourceRoot = ".";
 
   installPhase = ''
-    mkdir -p $out
-    cp -r bin $out/bin
-    chmod 555 $out/bin/excel_cmp
-    mv $out/bin/excel_cmp $out/bin/excel_cmp.sh
-    makeWrapper $out/bin/excel_cmp.sh $out/bin/excel_cmp \
-      --prefix PATH : ${lib.makeBinPath [ jdk ]}
+    install -Dt $out/share/excelcmp bin/dist/*
+    mkdir -p $out/bin
+    makeWrapper ${jre}/bin/java $out/bin/excelcmp \
+      --add-flags "-ea -cp $out/share/excelcmp/\* com.ka.spreadsheet.diff.SpreadSheetDiffer"
   '';
 
   meta = {
```
the trick is to prevent the wildcard `*` from being expanded by the shell